scripts/SetGateVariableTemplateCI.ps1

```diff
@@ -0,0 +1,46 @@
+param (
+    [Parameter(Mandatory=$true)]
+    [string]
+    $BaseImageName,
+
+    [Parameter(Mandatory=$true)]
+    [string]
+    $stackname,
+
+    [Parameter(Mandatory=$true)]
+    [string]
+    $VersionTextTemplateCI
+
+   )
+
+# #SetGateVariable.ps1 -BaseImageName w12r2d-14-2 -stackname BakingDP-ImageRelease
+# # Set the Gate variable if the file exists
+Write-Host "##vso[task.setvariable variable=IsEnabled;isOutput=true]False"
+$path = "$($env:Pipeline_Workspace)/_Build Image Release Artefacts/aws/$BaseImageName.txt"
+Write-Host "Using $path"
+if (Test-Path $path) {
+    try{
+        $amiID = Get-Content -Path $path
+        $amiID = $amiID.Split(" ")[0]   
+        Write-Host "AMI ID $($amiID)"
+        $imageName = (Get-EC2Image -ImageId $amiID).Name
+        $imageName -match "$BaseImageName[-]?[0-9]+"
+        $version = $Matches[0]
+        $version = $version -replace ".{2}$"  # Removes the last two elements from the string (which is "-8"). This will be replaced with 7 in the next step.
```

Comment:
What happens when the number becomes 2 digits?
I think an example value for this line
```
$version = $matches[0]
```
is `w19d-15-0-8`
another example is `w19d-15-0-89`, as the version will continue to be incremented.
So your code will be broken as soon as it becomes double figures.
The correct code is 
```
$version = $version -replace "-/d+$"
```
that replaces all the numbers at the end of the string that are prefixed with `-`.

Reply:
Ok, let me try and test this.

Reply:
_$version_ = $version -replace "-/d+$"
Above solution which you have suggested is also not working, giving wrong output. 

Tried below one
 _$version_ = $version -replace '-\d+$', ''    

Can use the above if we are absolutely sure that the first 9/10 characters won't change, we can simply slice them out instead of running regex.

Reply:
> `$version = $version -replace "-/d+$"` Above solution which you have suggested is also not working, giving wrong output.
> 
> Tried below one `$version = $version -replace '-\d+$', ''`
> 
> Can use the above if we are absolutely sure that the first 9/10 characters won't change, we can simply slice them out instead of running regex.

'/' was a typo

Use the regex.

Reply:
Required changes done, used below regex.

`$VersionLocal = $VersionLocal -replace "-\d+$"`